### PR TITLE
Add a `loadInvariant` builtin.

### PR DESCRIPTION
**Pull request by jckarter**
Lowers to an invariant load in LLVM; probably useful for SIL too at some point too, but NFC at that level yet.

### CODE REVIEW

**Comment by jckarter** [comment]
@swift-ci Please smoke test

File: include/swift/AST/DiagnosticsParse.def [hunk 1]
```diff
@@ -146,9 +146,7 @@ ERROR(lex_unary_equal,none,
       "'=' must have consistent whitespace on both sides", ())
 ERROR(extra_whitespace_period,none,
       "extraneous whitespace after '.' is not permitted", ())
-ERROR(lex_editor_placeholder,none,
-      "editor placeholder in source file", ())
-WARNING(lex_editor_placeholder_in_playground,none,
+WARNING(lex_editor_placeholder,none,
```
Comment (jrose-apple) [inline include/swift/AST/DiagnosticsParse.def]: This probably shouldn't have snuck in.

Comment (jckarter) [inline include/swift/AST/DiagnosticsParse.def]: Oops, thanks.

**Comment by jckarter** [comment]
@swift-ci Please smoke test

**Comment by jckarter** [comment]
@swift-ci Please smoke test

**Comment by jckarter** [comment]
@swift-ci Please smoke test

**Pull request merged**